hack/generator/pkg/jsonast/jsonast.go

```diff
@@ -295,6 +290,15 @@ func generateFieldDefinition(ctx context.Context, scanner *SchemaScanner, prop *
 		return field, nil
 	}
 
+	// This can happen if the property type was prune away by a type filter.
+	// There are a few options here: We can skip this property entirely, we can emit it
+	// with no type (won't compile), or we can emit with with interface{}.
+	// TODO: Currently setting this to anyType as that's easiest to deal with and will generate
+	// TODO: a warning during controller-gen
+	if propType == nil {
```

Comment:
I think a missing type should result in the property being omitted, with a log message at the [appropriate level](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md) ... maybe v2 or v3?

Emitting it with no type requires either manual fixes (yeuch!) or some other mechanism to individually suppress the affected properties. 

Using `interface{}` is problematic because controller gen doesn't really know what to do with it.

So I think suppressing the property, with visibility via logging, is the most viable approach.

Reply:
We're in a bit of a difficult situation with respect to logging here because a type `Person` can have a property of type `Hobby`, where `Hobby` is pruned, and then (later) the entire `Person` type can be excluded from emission.

The result is that... there really is no problem - we excluded the `Person` type from being emitted and `Hobby` was pruned as well. If we emit an error _right here_ though, we can't know if `Person` is going to be emitted or not. 

The reason I landed on `interface{}` was because:
1. It's a type filter not a property filter and I wasn't sure we wanted to conflate the two (note: There is no such thing as a property filter right now at least).
2. at controller-gen runtime, it sees what was actually emitted and so forcing a warning at that time solves the timing problem of the error.
3. controller-gen doesn't actually stop working if given a property with type `interface{}`, it just emits an error/warning and skips it.

Reply:
Perhaps we could attach warnings to types somehow (another map in scanner) and then only emit them if they don't get excluded later?

Reply:
But for now it should be fine to just emit a warning...

Reply:
Yeah, I was going to say what @Porges said.

Reply:
Created an issue to track the overall improvement (#138). Will fix behavior here to emit warning and skip property for now.